libraries/AP_LeakDetector/AP_LeakDetector.cpp

```diff
@@ -85,10 +110,16 @@ void AP_LeakDetector::init()
             _state[i].instance = i;
             _drivers[i] = new AP_LeakDetector_Digital(*this, _state[i]);
             break;
-#endif
+#else
         default:
-            _drivers[i] = NULL;
-            break;
+            if (_type[i] == 0) {
```

Comment:
Better to use the switch to set the defualt type. EG:

```
        switch (_pin[i]) {
#if (CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_CHIBIOS_FMUV3 || \
     CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_CHIBIOS_FMUV5)
        case 50 ... 55:
            _type[i].set_default(1)
            break;
        case 13 ... 15:
            _type[i].set_default(0)
            break;
#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_NAVIGATOR
        case 27:
            _type[i].set_default(1)
            break;
```
Then you can just switch on type as normal.

Reply:
@IamPete1 is there an advantage to using `set_default()` here instead of just `set()`?

Reply:
Set default will only change the param if the user has never set it (its new so they wont). It makes it possible to have a digital on pin 13 if the user configure it that way.

Reply:
Ah so `set_default()` gives users the option to change this param, i.e. providing more freedom on the supported boards rather than locking in their defaults with an overwrite. That makes sense to me!